### PR TITLE
Strong name signing.

### PR DESCRIPTION
**Pull request by livarcocc**
Adding a key to use for strong name signing of our assemblies and modifying the project.json of our projects to include that key.

### CODE REVIEW

**Comment by livarcocc** [comment]
@piotrpMSFT @brthor @Sridhar-MS

File: src/Microsoft.DotNet.Runtime/project.json [hunk 1]
```diff
@@ -1,7 +1,8 @@
 {
   "version": "1.0.0-*",
   "compilationOptions": {
-      "emitEntryPoint": true
+    "emitEntryPoint": true,
+    "keyFile": "../../tools/Key.snk"
```
Comment (Sridhar-MS) [inline src/Microsoft.DotNet.Runtime/project.json]: This project is just for publishing the runtime. You can skip signing this.

**Comment by Sridhar-MS** [comment]
LGTM except for one minor comment.

**Comment by davidfowl** [comment]
Can we Jair strong name true libraries not the exes

**Comment by livarcocc** [comment]
I put them on the exe to sign the dll versions of the executables as well. Is that not desirable then? If not, easy fix!

**Comment by davidfowl** [comment]
It's not. We only need to strong name the class libraries.

**Comment by livarcocc** [comment]
@davidfowl Done.

File: src/Microsoft.DotNet.Cli.Utils/project.json [hunk 2]
```diff
@@ -1,6 +1,10 @@
 ﻿{
     "version": "1.0.0-*",
 
+    "compilationOptions": {
```
Comment (davidfowl) [inline src/Microsoft.DotNet.Cli.Utils/project.json]: Can be removed as its shared source and gets compiled into other projects

**Comment by davidfowl** [comment]
One small change then it's good to go

**Comment by livarcocc** [comment]
@davidfowl Done. I will merge once all checks pass.

**Comment by livarcocc** [comment]
@dotnet-bot test OSX Release Build

**Comment by piotrpMSFT** [comment]
@dotnet-bot test OSX Release Build please

**Comment by livarcocc** [comment]
I guess I needed to be polite.

**Pull request merged**